netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,226 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.app
+
+import com.vladsch.flexmark.html.HtmlRenderer
+import com.vladsch.flexmark.parser.Parser
+
+import java.awt.{BorderLayout, Dimension, Graphics, GridBagConstraints, GridBagLayout, Insets}
+import java.awt.event.{ MouseAdapter, MouseEvent}
+import java.util.prefs.Preferences
+import javax.swing.{ JEditorPane, JLabel, JOptionPane, JPanel, JScrollPane, SwingConstants}
+import org.nlogo.app.infotab.InfoFormatter
+import org.nlogo.core.I18N
+import org.nlogo.swing.{HoverDecoration, OptionDialog}
+import org.nlogo.theme.{InterfaceColors, ThemeSync}
+
+import org.json.simple.parser.JSONParser
+import org.json.simple.{JSONArray, JSONObject}
+
+import scala.io.Source
+
+case class JsonObject(eventId: Int, date: String, title: String, fullText: String)
+
+class NotificationBanner() extends JPanel with ThemeSync with HoverDecoration {
+  private var jsonObjectList: List[JsonObject] = List() // Initialize here first
+  private val jsonUrl = "https://ccl.northwestern.edu/netlogo/announce-test.json"
+
+  jsonObjectList = parseJsonToList(fetchJsonFromUrl())
+  private var scrollPane = new JScrollPane()
+  private lazy val lastSeenEventIdKey: String = "lastSeenEventId" // The key for the most recently seen event-id
+  private var editorPane: JEditorPane = new JEditorPane()
```

Comment:
Also, why is this a JEditorPane if you can't edit it? Maybe choose something like JTextArea (org.nlogo.swing.TextArea) instead.

Reply:
This is to allow for formatting of the markdown into HTML

Reply:
Makes sense.